agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AzureMonitorLogProcessor.java

```diff
@@ -21,9 +23,16 @@ public void onEmit(Context context, ReadWriteLogRecord logRecord) {
     }
 
     ReadableSpan readableSpan = (ReadableSpan) currentSpan;
-
-    logRecord.setAttribute(
-        AiSemanticAttributes.OPERATION_NAME, OperationNames.getOperationName(readableSpan));
+    if (ConfigurationBuilder.inAzureFunctionsWorker()) {
+      AzureFunctionsCustomDimensions customDimensions =
+          AzureFunctionsCustomDimensions.fromContext(context);
+      if (customDimensions != null && customDimensions.operationName != null) {
```

Comment:
(you could leave this condition off, because `setAttribute` below is no-op with `null` value, the else condition below already relies on that behavior)
```suggestion
      if (customDimensions != null) {
```

Reply:
Kaibo tested it, operationName showed up correctly for logs from requests.. but metrics or other telemetry kind don't have it.  i suspect `AzureFunctionsCustomDimensions.fromContext(context)` context here is not the context from azure function, which used `ai-functions-custom-dimensions` as the context key.